Deploy to RBAC enabled cluster on Google Kubernetes Engine

Tasseri opened the pull request:
A manual to enable the Quick Full-Install Halyard Spinnaker deployment to be able to deploy images to other GCP projects with GKE that have RBAC enabled.

Tasseri committed:
add halyard-gke-deploy-rbac manual

Tasseri committed:
add to halyard quickstart list and remove extra title

Tasseri committed:
change link text for clarity and conformity

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+
+If you've deployed Spinnaker already using [this
+codelab](/setup/quickstart/halyard-gke) you're left with a Spinnaker that can only deploy to the cluster that Spinnaker is deployed in.
+In many cases you want to deploy to clusters in another GCP project that also have Kubernetes RBAC enabled.
```
"..in another GCP project that has Kubernetes RBAC enabled."

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+codelab](/setup/quickstart/halyard-gke) you're left with a Spinnaker that can only deploy to the cluster that Spinnaker is deployed in.
+In many cases you want to deploy to clusters in another GCP project that also have Kubernetes RBAC enabled.
+
+In this codelab, we will enable Spinnaker to push your containers to another GCP project that have a cluster secured by RBAC. This codelab will allow Spinnaker to be able to manipulate all deployments in the target cluster. 
```
“This codelab shows how to push your containers to a separate, RBAC-enabled cluster in a different GCP project, and how to enable Spinnaker to manipulate deployments in that cluster.”

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+In many cases you want to deploy to clusters in another GCP project that also have Kubernetes RBAC enabled.
+
+In this codelab, we will enable Spinnaker to push your containers to another GCP project that have a cluster secured by RBAC. This codelab will allow Spinnaker to be able to manipulate all deployments in the target cluster. 
+You may want to limit Spinnaker further but how to do that is not covered by this tutorial. 
```
Limit in what sense? Is this necessary here?

Tasseri replied:
Limit in the sense of that Spinnaker will have access to edit in all namespaces, added clarification for this.

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+
+In this codelab, we will enable Spinnaker to push your containers to another GCP project that have a cluster secured by RBAC. This codelab will allow Spinnaker to be able to manipulate all deployments in the target cluster. 
+You may want to limit Spinnaker further but how to do that is not covered by this tutorial. 
+This guide won't show you how to setup Spinnaker in a RBAC cluster.
```
"This guide doesn't show you to install Spinnaker in an RBAC-enabled cluster."

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+
+A GCP project (referred to as $GCP_PROJECT) running a Spinnaker [GKE cluster with halyard](/setup/quickstart/halyard-gke/)
+
+Another K8s cluster (referred to as $K8_TEST) running with Kubernetes version > 1.6 on another GCP project ($GCP_TEST)
```
Can you make bullet these two prereqs?

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
````diff
@@ -0,0 +1,159 @@
+gcloud info
+```
+
+If you are unfamiliar with how to use multiple project with gcloud you can use this [guide](https://cloud.google.com/sdk/docs/managing-configurations) to get started
````
s/project/projects
Also, format `gcloud`

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+
+Another K8s cluster (referred to as $K8_TEST) running with Kubernetes version > 1.6 on another GCP project ($GCP_TEST)
+
+## Part 1: Configure Gcloud
```
"gcloud" (lower-case g)

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+
+## Part 2: Add Service account to GCP
+
+Create the service account in GCP test project (note that you get the service account from your Spinnaker project)
```
"Create the service account in *the* GCP test project." I think the parenthetical note is not necessary because you have the comment in the code fragment below.

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
````diff
@@ -0,0 +1,159 @@
+kubectl apply -f ./spinnaker-service-account.yaml
+```
+
+Then we need to set access for Spinnaker to edit the cluster in order to let Spinnaker manage deployments to the cluster
````
"Then we need  to set access for Spinnaker to edit the cluster so it can manage deployments there."

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
````diff
@@ -0,0 +1,159 @@
+    --clusterrole edit
+```
+    
+If you can't create a clusterrolebinding make sure that you have the right permissions.
````
Format `clusterrolebinding`

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
```diff
@@ -0,0 +1,159 @@
+
+## Part 4: Add provider to halyard
+
+SSH in to your halyard instance
```
s/in to/into

dorbin replied:
Oh, and capitalize Halyard.

dorbin commented on setup/quickstart/halyard-gke-deploy-rbac/index.md:
````diff
@@ -0,0 +1,159 @@
+gcloud container clusters get-credentials $K8_TEST --project $GCP_TEST --zone us-central1-f
+```
+
+Get the new user profile that was created by gcloud and add the token you received before in part 3
````
`gcloud`

dorbin reviewed (commented):
Wilhelm, some editing suggestions. Feel free to push back if I break the meaning for any of this :)

Capitalize Kubernetes throughout.

I recommend numbering procedure steps. Some of them, like in Part  1, have only two steps, so it almost seems unnecessary, but I think it would be good to do so consistently in all those procedural sections.

Also, add periods at the ends of all these sentences :)

Thanks for the doc!
Dave

Tasseri committed:
fixes after dorbin input 1

Tasseri committed:
Merge branch 'master' of https://github.com/spinnaker/spinnaker.github.io

Tasseri committed:
fix numbering and halyard to upper-case

Tasseri committed:
fix whitespaces

Tasseri committed:
fix header out of place

Tasseri committed:
remove bash form inline formating

Tasseri commented:
Thank you for the feedback Dave!
I have worked with the issues you commented on.

I added the numbering as you requested. This is not used in the other guides on Spinnaker.io but I agree with you that it is better to have.

Leaving out the periods was a conscious decision since the page I used as a model was lacking them: https://www.spinnaker.io/setup/quickstart/halyard-gke/
But they are now added.

Ps. Sorry for all the commits didn't realise they would show up here. I am not used to work with an upstream PR ds.

Wilhelm

dorbin commented:
Thanks for the doc, Wilhelm, and the changes! Happy holidays!
